### PR TITLE
PYTHON-3757 Bulk write operations don't consider hint field in equality checks

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

**Review by ShaneHarvey — COMMENTED** [review]
Is this only a bug with DeleteOne or are there other operations we need to change?

File: test/test_crud_v1.py [hunk 1]
```diff
@@ -185,9 +185,21 @@ def test_InsertOneNotEquals(self):
 
     def test_DeleteOneEquals(self):
         self.assertEqual(DeleteOne({"foo": 42}), DeleteOne({"foo": 42}))
+        self.assertEqual(DeleteOne({"foo": 42}, {"bar": 43}), DeleteOne({"foo": 42}, {"bar": 43}))
```
Comment (ShaneHarvey) [inline test/test_crud_v1.py:188]: Could you use a minimal collation like `{"locale": "en_US"}` instead of "bar"?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

**Comment by NoahStapp** [comment]
> Is this only a bug with DeleteOne or are there other operations we need to change?

Good catch, I fixed DeleteMany as well!

**Review by ShaneHarvey — APPROVED** [review]
LGTM!

**Pull request merged**